pkg/avfoundation/avfoundation_callback_darwin.go

```diff
@@ -0,0 +1,53 @@
+package avfoundation
+
+// extern void onData(void*, void*, int);
+import "C"
+import (
+	"sync"
+	"unsafe"
+)
+
+var mu sync.Mutex
+var nextID handleID
+
+type dataCb func(data []byte)
+
+var handles = make(map[handleID]dataCb)
+
+type handleID int
+
+//export onData
+func onData(userData unsafe.Pointer, buf unsafe.Pointer, length C.int) {
+	data := C.GoBytes(buf, length)
+
+	handleNum := (*C.int)(userData)
+	cb := lookup(handleID(*handleNum))
```

Comment:
Looks better to handle `cb == nil` case. I guess it may happen right after `unregister`.

Reply:
Yea, but, I'm not sure the right behavior for this. Since this lookup is to give us back the session instance, if this failed to lookup, we can't return the error back to the session object.

Reply:
I think it may happens if the user unregister `handleID` and the data captured right before unregistering arrives here.
In this case `if cb == nil { return }` would be enough.

Reply:
Ah, I see what you mean. I'll add that. Thanks!